go/cmd/dolt/commands/show.go

```diff
@@ -186,181 +321,255 @@ func parseHashString(hashStr string) (hash.Hash, error) {
-		} else if upperCaseSpecRef == doltdb.Staged {
-			refHash, err = roots.Staged.HashOf()
-		} else {
-			refHash, err = parseHashString(specRef)
-		}
-		if err != nil {
-			return err
-		}
-		value, err := dEnv.DoltDB.ValueReadWriter().ReadValue(ctx, refHash)
-		if err != nil {
-			return err
-		}
-		if value == nil {
-			return fmt.Errorf("Unable to resolve object ref %s", specRef)
-		}
+func showCommit(queryist cli.Queryist, sqlCtx *sql.Context, opts *showOpts, commit *commitInfo) error {
 
-		if !opts.pretty {
-			cli.Println(value.Kind(), value.HumanReadableString())
-		}
+	cmHash := commit.commitHash
+	parents := commit.parentHashes
 
-		// If this is a commit, use the pretty printer. To determine whether it's a commit, try calling NewCommitFromValue.
-		commit, err := doltdb.NewCommitFromValue(ctx, dEnv.DoltDB.ValueReadWriter(), dEnv.DoltDB.NodeStore(), value)
+	cli.ExecuteWithStdioRestored(func() {
+		pager := outputpager.Start()
+		defer pager.Stop()
 
-		if err == datas.ErrNotACommit {
-			if !dEnv.DoltDB.Format().UsesFlatbuffers() {
-				return fmt.Errorf("dolt show cannot show non-commit objects when using the old LD_1 storage format: %s is not a commit", specRef)
-			}
-			cli.Println(value.Kind(), value.HumanReadableString())
-		} else if err == nil {
-			showCommit(ctx, dEnv, opts, commit)
-		} else {
-			return err
-		}
-	} else { // specRef is a CommitSpec, which must resolve to a Commit.
-		commitSpec, err := getCommitSpec(specRef)
-		if err != nil {
-			return err
-		}
+		showCommitInfo(pager, 0, opts.showParents, opts.decoration, commit)
+	})
 
-		err = showCommitSpec(ctx, dEnv, opts, commitSpec)
-		if err != nil {
-			return err
-		}
+	if len(parents) == 0 {
+		return nil
+	}
+	if len(parents) > 1 {
+		return fmt.Errorf("requested commit is a merge commit. 'dolt show' currently only supports viewing non-merge commits")
 	}
-	return nil
-}
-
-func showCommitSpec(ctx context.Context, dEnv *env.DoltEnv, opts *showOpts, commitSpec *doltdb.CommitSpec) error {
 
-	headRef, err := dEnv.RepoStateReader().CWBHeadRef()
-	if err != nil {
-		return err
+	parentHash := parents[0]
+	datasets := &diffDatasets{
+		fromRef: parentHash,
+		toRef:   cmHash,
 	}
 
-	commit, err := dEnv.DoltDB.Resolve(ctx, commitSpec, headRef)
+	// An empty string will cause all tables to be printed.
+	var tableNames []string
+
+	tableSet, err := parseDiffTableSetSql(queryist, sqlCtx, datasets, tableNames)
 	if err != nil {
 		return err
 	}
 
-	if opts.pretty {
-		err = showCommit(ctx, dEnv, opts, commit)
-		if err != nil {
-			return err
-		}
-	} else {
-		value := commit.Value()
-		cli.Println(value.Kind(), value.HumanReadableString())
+	dArgs := &diffArgs{
+		diffDisplaySettings: opts.diffDisplaySettings,
+		diffDatasets:        datasets,
+		tableSet:            tableSet,
 	}
-	return nil
+
+	return diffUserTables(queryist, sqlCtx, dArgs)
 }
 
-func showCommit(ctx context.Context, dEnv *env.DoltEnv, opts *showOpts, comm *doltdb.Commit) error {
```

Comment:
can you s/show/print/ for all of the show* commands. Hoping to get more consistent with stuff like this over time

Reply:
Will standardize the function names.